transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -458,22 +468,22 @@ public final ChannelHandlerInvoker invoker() {
         }
 
         @Override
-        public final ChannelOutboundBuffer outboundBuffer() {
+        public ChannelOutboundBuffer outboundBuffer() {
             return outboundBuffer;
         }
 
         @Override
-        public final SocketAddress localAddress() {
+        public SocketAddress localAddress() {
             return localAddress0();
         }
 
         @Override
-        public final SocketAddress remoteAddress() {
+        public SocketAddress remoteAddress() {
```

Comment:
Could you explain why you need to make this these methods non-final? localAddress0() and remoteAddress0() are abstract and already, so you don't need to make localAddress() and remoteAddress() non-final.